libs/util/data-fetcher/src/lib/readers/wfs.ts

```diff
@@ -0,0 +1,117 @@
+import { WfsEndpoint, WfsVersion } from '@camptocamp/ogc-client'
+import { DataItem, DatasetInfo, PropertyInfo } from '../model'
+import { fetchDataAsText } from '../utils'
+import { BaseReader } from './base'
+import { GmlReader, parseGml } from './gml'
+import { GeojsonReader, parseGeojson } from './geojson'
+
+export class WfsReader extends BaseReader {
+  endpoint: WfsEndpoint
+  featureTypeName: string
+  version: WfsVersion
+
+  constructor(url: string, wfsEndpoint: WfsEndpoint, featureTypeName: string) {
+    super(url)
+    this.endpoint = wfsEndpoint
+    this.featureTypeName = featureTypeName
+    this.version = this.endpoint.getVersion()
+  }
+
+  get properties(): Promise<PropertyInfo[]> {
+    return this.getData().then((result) => result.properties)
+  }
+
+  get info(): Promise<DatasetInfo> {
+    return this.getData().then(
+      (result) =>
+        ({
+          itemsCount: result.items.length,
+        }) as DatasetInfo
+    )
```

Comment:
I think this is supposed to be the total amount of items in the featureType, you should be able to get this information using ogc-client:
```ts
const count = (await this.endpoint.getFeatureTypeFull(featureTypeName)).objectCount
```

Reply:
Right! i have updated the code like you have suggested.